cmd/lakectl/cmd/sst.go

```diff
@@ -0,0 +1,180 @@
+package cmd
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"time"
+
+	sst "github.com/cockroachdb/pebble/sstable"
+	"github.com/spf13/cobra"
+	"github.com/treeverse/lakefs/catalog/rocks"
+	"github.com/treeverse/lakefs/graveler/committed"
+	"github.com/treeverse/lakefs/graveler/sstable"
+)
+
+type InMemFileInfo struct {
+	name string
+	size int64
+}
+
+func (r *InMemFileInfo) Name() string       { return r.name }
+func (r *InMemFileInfo) Size() int64        { return r.size }
+func (r *InMemFileInfo) Mode() os.FileMode  { return os.ModePerm }
+func (r *InMemFileInfo) ModTime() time.Time { return time.Now() }
+func (r *InMemFileInfo) IsDir() bool        { return false }
+func (r *InMemFileInfo) Sys() interface{}   { return nil }
+
+type InMemReadableFile struct {
+	data *bytes.Reader
+}
+
+func (r *InMemReadableFile) ReadAt(p []byte, off int64) (n int, err error) {
+	return r.data.ReadAt(p, off)
+}
+
+func (r *InMemReadableFile) Close() error {
+	return nil
+}
+
+func (r *InMemReadableFile) Stat() (os.FileInfo, error) {
+	return &InMemFileInfo{
+		name: "stdin",
+		size: r.data.Size(),
+	}, nil
+}
+
+func getIterFromFile(filePath string) (committed.ValueIterator, map[string]string, error) {
+	var file sst.ReadableFile
+	var err error
+	// read from stdin (file has to be seekable/stat-able so we have this weird wrapper
+	if filePath == "-" {
+		data, err := ioutil.ReadAll(os.Stdin)
```

Comment:
This is scary because of memory consumption.  But there is no reason to do that.
1. Many times standard input *will* be seekable, just try.
2. When standard input isn't seekable, copy it to a temporary file and read _that_.